src/cephadm/cephadm

```diff
@@ -3074,22 +3087,31 @@ def command_deploy():
     else:
         logger.info('%s daemon %s ...' % ('Deploy', args.name))
 
+    # Get and check ports explicitly required to be opened
+    daemon_ports = [] # type: List[int]
+    if args.tcp_ports:
+        daemon_ports = list(map(int, args.tcp_ports.split()))
+        if any([port_in_use(port) for port in daemon_ports]):
+            raise Error("TCP Port(s) '{}' required for {} already in use".format(",".join(map(str, daemon_ports)), daemon_type))
```

Comment:
As you have added the ports to `deploy_daemon`, can you move this check there? Otherwise we'd have this check four times in `command_deploy`.